labs/unicorn-store/software/alternatives/unicorn-store-quarkus/src/main/java/com/unicorn/store/data/UnicornRepository.java

```diff
@@ -0,0 +1,40 @@
+package com.unicorn.store.data;
+
+import com.unicorn.store.model.Unicorn;
+import com.unicorn.store.model.UnicornEventType;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.transaction.Transactional;
+import java.util.Optional;
+
+@ApplicationScoped
+public class UnicornRepository {
```

Comment:
Let's use var entity instead of Unicorn entity

Reply:
That doesn't work as it is `PanacheEntityBase` and not `Unicorn`.